src/main/java/org/bytedeco/javacpp/tools/Parser.java

```diff
@@ -3774,18 +3865,22 @@ boolean group(Context context, DeclarationList declList) throws ParserException
                     constructors += "    /** Pointer cast constructor. Invokes {@link Pointer#Pointer(Pointer)}. */\n" +
                                  "    public " + shortName + "(Pointer p) { super(p); }\n";
                 }
-                if (defaultConstructor && (info == null || !info.purify) && (!abstractClass || ctx.virtualize) && !arrayConstructor) {
-                    constructors += "    /** Native array allocator. Access with {@link Pointer#position(long)}. */\n" +
-                                 "    public " + shortName + "(long size) { super((Pointer)null); allocateArray(size); }\n" +
-                                 "    private native void allocateArray(long size);\n" +
-                                 "    @Override public " + shortName + " position(long position) {\n" +
-                                 "        return (" + shortName + ")super.position(position);\n" +
-                                 "    }\n" +
-                                 "    @Override public " + shortName + " getPointer(long i) {\n" +
-                                 "        return new " + shortName + "((Pointer)this).offsetAddress(i);\n" +
-                                 "    }\n";
-                }
+                addArrayConstructor =  (defaultConstructor && (info == null || !info.purify) && (!abstractClass || ctx.virtualize)
+                    && !arrayConstructor && constructorAnnotations.isEmpty());
+            }
+
+            if (addArrayConstructor) {
+                constructors += "    /** Native array allocator. Access with {@link Pointer#position(long)}. */\n" +
+                                "    public " + shortName + "(long size) { super((Pointer)null); allocateArray(size); }\n" +
+                                "    private native void allocateArray(long size);\n" +
+                                "    @Override public " + shortName + " position(long position) {\n" +
+                                "        return (" + shortName + ")super.position(position);\n" +
+                                "    }\n" +
+                                "    @Override public " + shortName + " getPointer(long i) {\n" +
+                                "        return new " + shortName + "((Pointer)this).offsetAddress(i);\n" +
+                                "    }\n";
```

Comment:
Please do that in a way that doesn't change the output for the presets.

Reply:
I rearranged so that the result is closer to what master outputs. But there is still a difference because in master the array constructor and pointer cast constructor are not emitted in the same order whether `implicitConstructor` is true or not.
Do you prefer to keep this version or a version that reproduces exactly the master output but with a uglier code with repetitions ?

Reply:
We don't need repetitions, just declare a like `String arrayConstructor = ...` instead of a boolean flag and use that in your modifications.

Reply:
In master, array constructor  and `allocateArray` are interleaved with other lines when `implicitConstructor` is true, so I don't think it would help.
I reverted to master and just added the tests for `constructorAnnotations.isEmpty()`. The parser output is now the same as for master when there are no annotations on constructors. If you think of a nicer way to implement this, feel free to modify yourself.

Reply:
Seems a lot better, but could you revert unnecessary changes in formatting?

Reply:
I guessed you mean the alignment of + with = in concatenations. Good enough now ?